typify-impl/src/enums.rs

```diff
@@ -609,28 +609,47 @@ impl TypeSpace {
             // account for types such as Uuid whose names come from outside of
             // the schema... but you can't win them all.
             .map(schema_is_named)
-            .collect::<Option<Vec<_>>>()
-            // Prune the common prefixes from all variant names. If this
-            // results in any of them being empty, we don't use these names.
-            .and_then(|variant_names| {
-                let common_prefix = variant_names
-                    .iter()
-                    .cloned()
-                    .reduce(|a, b| get_common_prefix(&a, &b))
-                    .unwrap();
-                variant_names
-                    .into_iter()
-                    .map(|var_name| {
-                        let var_name = &var_name[common_prefix.len()..];
-                        if var_name.is_empty() {
-                            None
-                        } else {
-                            Some(var_name.to_string())
-                        }
-                    })
-                    .collect::<Option<Vec<_>>>()
+            .collect::<Vec<_>>();
+
+        let common_prefix = variant_names
+            .iter()
+            .flatten()
+            .cloned()
+            .reduce(|a, b| get_common_prefix(&a, &b))
+            .unwrap_or_else(String::new);
```

Comment:
```suggestion
            .unwrap_or_default();
```

Does this work the same?